http2/src/test/java/com/blazemeter/jmeter/http2/sampler/HTTP2StreamHandlerTest.java

```diff
@@ -239,4 +238,24 @@ public void onDataTest2() {
     assertTrue(resSR.isSuccessful());
   }
 
-} 
+  @Test
+  public void onResetTest() {
+
+    ResetFrame resetFrame = new ResetFrame(0, ErrorCode.REFUSED_STREAM_ERROR.code);
+    http2SampleResult = buildThreadVarsResult();
+    http2StreamHandler = new HTTP2StreamHandler(http2Connection, url, null,
+        null, http2SampleResult);
+    http2SampleResult.sampleStart();
+    http2StreamHandler.onReset(stream, resetFrame);
+    http2SampleResult = http2StreamHandler.getHTTP2SampleResult();
+    assertEquals(String.valueOf(ErrorCode.REFUSED_STREAM_ERROR.code), http2SampleResult.getResponseCode());
+  }
+
+  public HTTP2SampleResult buildThreadVarsResult(){

```

Comment:
consider using this method on other methods like onPushTest and onDataTest, and removing the redundant setup of threadVars in onHeadersTest